# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [5]:
import pandas as pd
import numpy as np
from scipy import stats

In [6]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [14]:
# number of callbacks for black-sounding names
print sum(data[data.race=='b'].call),"callbacks for black-sounding names out of", data[data.race=='b'].id.count()
print sum(data[data.race=='w'].call),"callbacks for white-sounding names out of", data[data.race=='w'].id.count()

157.0 callbacks for black-sounding names out of 2435
235.0 callbacks for white-sounding names out of 2435


In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## Does the CLT hold?
Indipendence: Sample observations must be independent.
Sample size and skew.
These conditions are met CLT can be applied.
## Null Hypothesis
Race does not have a significant impact on the rate of callbacks for resumes. Population(white)=Population(black)
## Alternative Hypothesis
Race does have a significant impact on the rate of callbacks for resumes. Population(white)!=Population(black)

In [15]:

# First of all separate the resumes by race: b and w.
df_white=data[data.race=='w']
df_black=data[data.race=='b']

#Number of CV per race:
w_cvs=len(df_white.race)
b_cvs=len(df_black.race)

#Number of calls per race:
w_calls=sum(data[data.race=='w'].call)
b_calls=sum(data[data.race=='b'].call)

# Sample proportions:
w_sample_p = w_calls / w_cvs
b_sample_p = b_calls / b_cvs

print w_sample_p,b_sample_p

0.0965092402464 0.064476386037


In [16]:
ppool= round((w_calls+b_calls)/(w_cvs+b_cvs),2)
ppool

0.08

In [17]:
# Then let's check if n1p_pool >=10, n1(1-p_pool) >=10:

np_w=w_cvs*ppool
n1p_w=w_cvs*(1-ppool)

# and for n2p_pool >=10, n2(1-p_pool) >=10:
np_b=b_cvs*ppool
n1p_b=b_cvs*(1-ppool)

print np_w,n1p_w, np_b,n1p_b

194.8 2240.2 194.8 2240.2


How probable is it given the null hypothesis of no difference between populations and the expected number of call backs to be 195 to take a sample and get 157 callbacks for black-sounding names out of 2435, and 235.0 callbacks for white-sounding names out of 2435?

In [21]:
SE_diff=round((((ppool*(1-ppool))/b_cvs)+((ppool*(1-ppool))/w_cvs))**0.5,2)
Zvalue= round((w_sample_p - b_sample_p)/SE_diff,2)
print Zvalue
ME=round(Zvalue*SE_diff,2)
print ME

3.2
0.03


In [22]:
#confidence interval
CI_hi=ppool+ME
CI_lo=ppool-ME
print CI_lo, "-",CI_hi

0.05 - 0.11


Both population proportions found above fall within the CI in the cell directly above. There is no statistically significant difference. We can not reject the null hypothesis.